python/events/src/hydrotools/events/models/interception.py

```diff
@@ -0,0 +1,376 @@
+"""
+============
+Interception
+============
+
+Methods to support the estimation of rainfall interception for a single event
+across a single catchment. These methods implement the Gash model of canopy
+interception adjusted for a single storm using reasonable defaults from
+Miralles et al, 2010. The primary method in this module is
+`compute_interception`. Use this method to estimate intercepted rainfall on
+an event basis using default parameters from the literature.
+`compute_interception` will call the neccessary support methods in the
+appropriate order to estimate canopy interception per Gash (1979).
+
+Gash, J. H. C. (1979). An analytical model of rainfall interception by forests.
+    Quarterly Journal of the Royal Meteorological Society, 105(443), 43-55.
+Miralles, D. G., Gash, J. H., Holmes, T. R. H., De Jeu, R. A. M., & Dolman,
+    A. J. (2010). Global canopy interception from satellite observations.
+    Journal of Geophysical Research: Atmospheres, 115(D16), 2009JD013530.
+    https://doi.org/10.1029/2009JD013530
+Valente, F., David, J. S., & Gash, J. H. C. (1997). Modelling interception loss
+    for two sparse eucalypt and pine forests in central Portugal using
+    reformulated Rutter and Gash analytical models. Journal of hydrology,
+    190(1-2), 141-162.
+
+Functions
+---------
+compute_canopy_saturation
+compute_trunk_saturation
+compute_canopy_loss
+compute_trunk_loss
+compute_interception
+"""
+
+import warnings
+import numpy as np
+
+def compute_canopy_saturation(
+        rainfall_rate: float,
+        canopy_storage: float,
+        trunk_evaporation: float,
+        evaporation_rate: float
+) -> float:
+    """
+    Compute the amount of rainfall necessary to saturate the canopy.
+    Notated as $P_g'$ in Miralles et al, 2010. Typically in mm.
+
+    Parameters
+    ----------
+    rainfall_rate: float, required
+        Mean rainfall rate in L/T, typically mm/hour. Notated 'R'.
+    canopy_storage: float, required
+        Minimum depth of water required to saturate the canopy. Typically 1.2
```

Comment:
if this parameter has a "typical" value with little expected variance, could it be made an optional arg with 1.2 the default?

Reply:
Yeah, I chose to make these optional for the top level method, but required for the lower level methods. I figured it would be easier to maintain that way. I guess I could add another layer (module or object) to store the default values, then point all these parameters at that.

Reply:
I added a module level `DefaultParameter` dataclass.